client/pages/fight.jsx

```diff
@@ -1,8 +1,123 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useLocation } from 'react-router';
 import TypingBox from '../components/typing-box';
+import Countdown from '../components/countdown';
+import HPBar from '../components/hp-bar';
+import { Grid, Box, styled } from '@material-ui/core';
+import { connectSocket, disconnectSocket, getRandom, finishPhrase } from '../lib/fight-socket';
+import FightContext from '../lib/fight-context';
+
+const dummyMeta = {
+  gameId: 1,
+  hostId: 1,
+  hostUsername: 'Player 1',
+  oppId: 2,
+  oppUsername: 'Player 2'
+};
+
+const SpriteDummy = styled('div')({
+  backgroundColor: 'blue',
+  width: '20rem',
+  height: '30rem'
+});
 
 export default function Fight(props) {
+  const [metaData, setMetaData] = useState(null);
+  const location = useLocation();
+
+  const [counting, setCounting] = useState(false);
+  const [yourHp, setYourHp] = useState(100);
+  const [oppHp, setOppHp] = useState(100);
+  const [phrase, setPhrase] = useState('Getting phrase');
+
+  const gameId = location.search.replace('?gameId=', '');
+  const contextValue = { youFinishFirst: () => { finishPhrase(gameId, 1); } };
+
+  function removeCountdown() {
+    setCounting(false);
+  }
+
+  function damage(player) {
+    const hit = 20;
+    if (player === 'you') {
+      setYourHp(yourHp - hit);
+    } else if (player === 'opp') {
+      setOppHp(oppHp - hit);
+    }
+  }
+
+  // get metaData
+  useEffect(() => {
+    setMetaData(dummyMeta);
+  }, []);
+
+  // socket connection
+  useEffect(() => {
+    connectSocket(gameId, { setPhrase });
+    getRandom(gameId);
+    return () => {
+      disconnectSocket();
+    };
+  }, []);
+
+  if (!metaData) {
+    return <></>;
+  }
   return (
-    <TypingBox text="Type Text Here" />
+    <FightContext.Provider value={contextValue}>
+      <Grid container direction="column">
```

Comment:
Maybe just set a max width on this entire container and change the width values of other components to be percent-based over vw so the fight page components don't go past a certain width.